### PR TITLE
Added synchronisation messages.

### PR DESCRIPTION
**Pull request by BeeHiveJava**
Added support for identifying cars by Player. Also added support for a System message in the chat.

### CODE REVIEW

File: Soccar [UI]/src/nl/socnet/message/PlayerSyncMessage.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    private final float x;
+    private final float y;
+    private final float angle;
+
+    private final float linearVelocityX;
+    private final float linearVelocityY;
+
+    private final float angularVelocity;
+
+    public PlayerSyncMessage(int playerId, float x, float y, float angle, float linearVelocityX, float linearVelocityY, float angularVelocity) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/PlayerSyncMessage.java:22]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/PlayerSyncMessage.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
+package nl.socnet.message;
+
+import nl.soccar.socnet.message.Message;
+import nl.soccar.socnet.message.MessageConstants;
+import nl.soccar.socnet.message.MessageEvent;
+import nl.socnet.message.handler.PlayerSyncMessageHandler;
+
+@MessageEvent(id = MessageConstants.SYNC_POSITION_PLAYER_MESSAGE_ID, handler = PlayerSyncMessageHandler.class)
+public final class PlayerSyncMessage extends Message {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/PlayerSyncMessage.java:9]: MINOR Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/BallSyncMessage.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+    private final float x;
+    private final float y;
+
+    private final float linearVelocityX;
+    private final float linearVelocityY;
+
+    private final float angularVelocity;
+
+    public BallSyncMessage(float x, float y, float linearVelocityX, float linearVelocityY, float angularVelocity) {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/BallSyncMessage.java:19]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/BallSyncMessage.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package nl.socnet.message;
+
+import nl.soccar.socnet.message.Message;
+import nl.soccar.socnet.message.MessageConstants;
+import nl.soccar.socnet.message.MessageEvent;
+import nl.socnet.message.handler.BallSyncMessageHandler;
+
+@MessageEvent(id = MessageConstants.SYNC_POSITION_BALL_MESSAGE_ID, handler = BallSyncMessageHandler.class)
+public final class BallSyncMessage extends Message {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/BallSyncMessage.java:9]: MINOR Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+import io.netty.buffer.ByteBuf;
+import nl.soccar.physics.GameEngine;
+import nl.soccar.socnet.connection.Connection;
+import nl.soccar.socnet.message.MessageHandler;
+import nl.soccar.ui.Main;
+import nl.soccar.ui.fx.controller.GameViewFXMLController;
+import nl.socnet.message.BallSyncMessage;
+
+public final class BallSyncMessageHandler extends MessageHandler<BallSyncMessage> {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/handler/BallSyncMessageHandler.java:11]: MINOR Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import nl.soccar.socnet.connection.Connection;
+import nl.soccar.socnet.message.MessageHandler;
+import nl.soccar.ui.Main;
+import nl.soccar.ui.fx.controller.GameViewFXMLController;
+import nl.soccar.ui.rmi.ClientController;
+import nl.socnet.message.PlayerSyncMessage;
+
+import java.util.Optional;
+
+public final class PlayerSyncMessageHandler extends MessageHandler<PlayerSyncMessage> {
```
Comment (ghost) [inline Soccar [UI]/src/nl/socnet/message/handler/PlayerSyncMessageHandler.java:16]: MINOR Document this public class. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 6 issues
* MINOR 6 minor

Watch the comments in this conversation to review them.

File: Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java [hunk 7]
```diff
@@ -38,8 +36,19 @@
...
     private Canvas canvas;
 
+    private GameCanvas gameCanvas;
     private Color textColor;
 
+    private static float getAngle(float sourceX, float sourceY, float targetX, float targetY) {
+        float angle = (float) Math.toDegrees(Math.atan2(targetY - sourceY, targetX - sourceX));
+
+        if (angle < 0) {
+            angle += 360;
```
Comment (Laugslander) [inline Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java:46]: Is het mogelijk om dit met ternary operation op te lossen?

**Review by Laugslander — CHANGES_REQUESTED** [review]

File: Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java (hunk 7 above)
Comment (BeeHiveJava) [reply Soccar [UI]/src/nl/soccar/ui/fx/controller/GameViewFXMLController.java:46]: return angle < 0 ? angle + 360 : angle?

**Review by Laugslander — APPROVED** [review]

**Pull request merged**